### PR TITLE
Update binding on role change

### PR DESCRIPTION
**Pull request by corest**
Towards https://github.com/giantswarm/giantswarm/issues/10836

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: service/controller/resource/namespaceauth/create.go [hunk 1]
```diff
@@ -29,7 +29,7 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 	}
 	tenantAdminRole := role{
 		name:        "tenant-admin",
-		targetGroup: r.namespaceAuth.ViewAllTargetGroup,
+		targetGroup: r.namespaceAuth.TenantAdminTargetGroup,
```
Comment (kopiczko) [inline service/controller/resource/namespaceauth/create.go:32]: Bug? Should be mentioned in the changelog?

File: service/controller/resource/namespaceauth/create.go [hunk 2]
```diff
@@ -83,6 +83,16 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
...
+
+			if role.targetGroup != existingRoleBinding.Subjects[0].Name {
+				r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("updating role binding %#q", newRoleBinding.Name))
+				_, err := r.k8sClient.RbacV1().RoleBindings(namespace.Name).Update(newRoleBinding)
+				if err != nil {
+					return microerror.Mask(err)
+				}
+				r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("role binding %#q has been updated", newRoleBinding.Name))
+
+			}
```
Comment (kopiczko) [inline service/controller/resource/namespaceauth/create.go:95]: I'm not blocking but I'd like to have a nicer flow here to make future changes nice.

Suggestion:

```go
} else if needsUpdate(role, existingRoleBinding) {
	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("updating role binding %#q", newRoleBinding.Name))
	_, err := r.k8sClient.RbacV1().RoleBindings(namespace.Name).Update(newRoleBinding)
	if err != nil {
		return microerror.Mask(err)
	}
	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("role binding %#q has been updated", newRoleBinding.Name))

} else {
	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("role binding %#q already exists", newRoleBinding.Name))
}

func needsUpdate(role, existingRoleBinding) bool {
	role.targetGroup != existingRoleBinding.Subjects[0].Name
}
```

Then in the future if we decide update conditions change we need to update only `needsUpdate` func w/o changing this big if/else.

Comment (corest) [reply service/controller/resource/namespaceauth/create.go:95]: smart!

File: service/controller/resource/namespaceauth/create.go (hunk 1 above)
Comment (corest) [reply service/controller/resource/namespaceauth/create.go:33]: done

**Pull request merged**